### ISSUE TITLE
SingleFile doens't repect overridden custom filenames before converting to annotation mode

### ISSUE THREAD

**Issue by directentis1** [gildas-lormeau/SingleFile#1499]
**Describe the bug**

When I click SingleFile icon on the toolbar, it's prompted me for a custom filename (as my setting).
I set a name and click "OK" before it's switches to annotate page.
However, the name of the file is the default and cannot be overwritten.

When I press the save button after finishing editing, annotating on the page, the file name is as default instead of the name I customized.

**To Reproduce**
1. Go to a normal webpage
2. Click on SF's Icon on Firefox Toolbar
3. Have prompts to input custom filename, modify the default and click OK.
4. Annotating.
5. Click Save symbol after done.
6. Check file saved on filesystem or in Github Repository. It's as a default name defined by filename template, not custom one.

**Expected behavior**
Custom filename that has prompted user to input must priority than default template filename.

**My Configuration**
[singlefile-settings.json](https://github.com/user-attachments/files/16185817/singlefile-settings-2024-07-12T00_25_34.425Z.json)

**Environment**
 - OS: Ubuntu 22.04
 - Browser: Firefox
 - Version: 128.0 (64-bit)

**Additional context**
Sometime it works in a random way?!
But most of time it doesn't.

**Comment by directentis1** [comment]
> Thank you. From memory, it's due to a technical limitation of the extensions. I'll have to dig up the subject again, though, as the question has more or less already been asked, but I can't remember what problems I found. I'm surprised it works sometimes...

I curious if it's some kind of race condition... 
hmm, maybe setting the wait time a little longer would solve the problem?

**Comment by gildas-lormeau** [comment]
I did some refactoring and fixed this issue. I also made sure that the prompt dialog will be opened in the editor instead of before opening it. The fix will be available in the next version.

**Comment by gildas-lormeau** [comment]
There were some remaining bugs related to this issue. Hopefully, it should work as expected in the next version (publication in progress).

**Comment by directentis1** [comment]
@gildas-lormeau 

I got it.  The file name form embedded in saved html files is removed when using the custom name prompt option.

 Like the configuration file I attached above, the html files should be saved in the structure `<year>/<month>/<timestamp>` corresponding to the read date of the file (although I customized the file title to make it easier to remember, 

or often use the publication date of the relevant pages instead of the access time), but it saves directly to the root "/" whether saving to the Github Repository or saving to the local filesystem.
